### PR TITLE
Add dual reward distribution

### PR DESCRIPTION
**Pull request by Flocqst**

### CODE REVIEW

**Review by JaredBorders — CHANGES_REQUESTED** [review]
⏳ Preliminary Review Before Testing

File: contracts/StakingRewardsV2.sol [hunk 1]
```diff
@@ -602,6 +660,25 @@ contract StakingRewardsV2 is
         emit RewardAdded(_reward);
     }
 
+    /// @inheritdoc IStakingRewardsV2
+    function notifyUsdcRewardAmount(uint256 _reward)
```
[the remark this answers is not in the document]
Comment (Flocqst) [reply contracts/StakingRewardsV2.sol:664]: Merged into `notifyUsdcRewardAmount` so no longer an issue

File: contracts/StakingRewardsV2.sol [hunk 2]
```diff
@@ -602,6 +660,25 @@ contract StakingRewardsV2 is
...
+            rewardRate = _reward / rewardsDuration;
+        } else {
+            uint256 remaining = periodFinish - block.timestamp;
+            uint256 leftover = remaining * rewardRate;
+            rewardRate = (_reward + leftover) / rewardsDuration;
+        }
+
+        lastUpdateTime = block.timestamp;
+        periodFinish = block.timestamp + rewardsDuration;
+        emit UsdcRewardAdded(_reward);
```
[the remark this answers is not in the document]
Comment (Flocqst) [reply contracts/StakingRewardsV2.sol]: Need to change this into a single event with the notifier changes

File: contracts/StakingRewardsV2.sol [hunk 3]
```diff
@@ -377,6 +410,7 @@ contract StakingRewardsV2 is
     function _updateReward(address _account) internal {
```
[the remark this answers is not in the document]
Comment (Flocqst) [reply contracts/StakingRewardsV2.sol:410]: We discussed this yesterday, there is no need for separate functions as both will be notified together.

File: contracts/StakingRewardsV2.sol [hunk 4]
```diff
@@ -91,6 +91,24 @@ contract StakingRewardsV2 is
     /// @notice tracks all addresses approved to take actions on behalf of a given account
     mapping(address => mapping(address => bool)) public operatorApprovals;
 
+    /// @notice Contract for USDC ERC20 token - used for rewards
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    IERC20 public immutable usdc;
```
[the remark this answers is not in the document]
Comment (JaredBorders) [reply contracts/StakingRewardsV2.sol:96]: @Flocqst still needs to be moved

File: contracts/StakingRewardsNotifier.sol [hunk 5]
```diff
@@ -83,6 +83,6 @@ contract StakingRewardsNotifier is Ownable2Step, IStakingRewardsNotifier {
         delete mintedAmount;
```
Comment (JaredBorders) [inline contracts/StakingRewardsNotifier.sol:83]: please remove this and remove the `mintedAmount` as we discussed

**Review by JaredBorders — APPROVED** [review]
Nice work 🫡

Please make the changes to the Notifier and add the documentation detailing "what its purpose is" (as we discussed last week).

following that, it is ready for audit.

File: contracts/StakingRewardsV2.sol (hunk 4 above)
[the remark this answers is not in the document]
Comment (Flocqst) [reply contracts/StakingRewardsV2.sol:96]: Moved

File: contracts/StakingRewardsV2.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (Flocqst) [reply contracts/StakingRewardsV2.sol]: Changed

File: contracts/StakingRewardsNotifier.sol (hunk 5 above)
Comment (Flocqst) [reply contracts/StakingRewardsNotifier.sol]: Removed, moved comments saying why we have to keep `mintedAmount` in the function signature to the interface NatSpec.

Comment (Flocqst) [reply contracts/StakingRewardsNotifier.sol]: Also added usdc distribution to the notifier.

**Pull request merged**